src/snitch_append.cpp

```diff
@@ -81,23 +88,23 @@ bool append_fast(small_string_span ss, const void* ptr) noexcept {
     if (ptr == nullptr) {
         return append(ss, nullptr);
     } else {
-        return append_fmt(ss, ptr);
+        return append(ss, "0x") && append_to_chars(ss, reinterpret_cast<std::size_t>(ptr), 16);
```

Comment:
Technically more correct, although should generally be the same. This might require `#include <cstdint>`.
```suggestion
        return append(ss, "0x") && append_to_chars(ss, reinterpret_cast<std::uintptr_t>(ptr), 16);
```

Reply:
good catch, I have some older products on old MCUs where these are not the same. I must be getting spoiled by these 32-bit ARMs 😄